### DISCUSSION TITLE
No response when using GDAL layer(imagery and elevation)

### DISCUSSION THREAD

**Discussion by wwwronin** [pelicanmapping/rocky#1] [Q&A / Support]
I try the rocky-demo, and there is no response when using GDAL layers . Here is screenshot.
![image](https://user-images.githubusercontent.com/6395346/233832078-5843d67c-33c5-4d6a-8981-391fe2e8e7ba.png)

**Comment by gwaldron** [comment]
Hello, can you provide more detailed information please?

**Comment by wwwronin** [comment]
I use vcpkg as build solution, VS2022 x64 bit compiler, Windows 10 Professional ,64bit. Here is the code I used in demo .
int main(int argc, char** argv)
{
    // instantiate the engine.
    rocky::EngineVSG engine(argc, argv);

    // add an imagery layer
    auto imagery = rocky::GDALImageLayer::create(); // rocky::TMSImageLayer::create();
    imagery->setURI("./data/world.tif");
    engine.map()->layers().add(imagery);

    if (imagery->status().failed())
        return error(imagery->status());

    // add an elevation layer
    auto elevation = rocky::GDALElevationLayer::create();
    elevation->setURI("./data/china_dem.tif");
    engine.map()->layers().add(elevation);

    if (elevation->status().failed())
        return error(elevation->status());

    // run until the user quits.
    return engine.run();
}

**Comment by gwaldron** [comment]
Is the problem the bump on the right in your screenshot? Can you provide your elevation tif, or post the output of gdalinfo china.tif?

**Comment by wwwronin** [comment]
Not only bump on the right ,the program seems no response(dead), I wait about 10 minutes, still no response.Here is the output of elevation file .
Driver: GTiff/GeoTIFF

Size is 15840, 7920
Coordinate System is:
GEOGCRS["WGS 84",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433]],
    CS[ellipsoidal,2],
        AXIS["latitude",north,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433]],
        AXIS["longitude",east,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433]],
    ID["EPSG",4326]]
Data axis to CRS axis mapping: 2,1
Origin = (70.000000000000000,53.000000000000000)
Pixel Size = (0.004166666666667,-0.004166666666667)
Metadata:
  AREA_OR_POINT=Area
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  (  70.0000000,  53.0000000) ( 70d 0' 0.00"E, 53d 0' 0.00"N)
Lower Left  (  70.0000000,  20.0000000) ( 70d 0' 0.00"E, 20d 0' 0.00"N)
Upper Right ( 136.0000000,  53.0000000) (136d 0' 0.00"E, 53d 0' 0.00"N)
Lower Right ( 136.0000000,  20.0000000) (136d 0' 0.00"E, 20d 0' 0.00"N)
Center      ( 103.0000000,  36.5000000) (103d 0' 0.00"E, 36d30' 0.00"N)
Band 1 Block=15840x1 Type=Int16, ColorInterp=Gray
  NoData Value=nan

**Comment by gwaldron** [comment]
I assume it's fine with just imagery?
Can you attach your elevation geotiff?

**Reply by wwwronin** [reply]
I remove both elevation and imagery layer, the rocky-demo still no response when loading .

**Comment by wwwronin** [comment]
Try just only the imagery, still dead . I think there may have bugs in GDAL layer. Due to the size limitition, I cut a aera for you .
[ex.zip](https://github.com/pelicanmapping/rocky/files/11303768/ex.zip)

**Comment by gwaldron** [comment]
Thanks. Using your tif I was able to fix the elevation reader to make it work correctly.
However, I cannot reproduce the "lock up" problem you are seeing.

**Reply by wwwronin** [reply]
You are welcome. Lock up problem is caused of OS ,I switch the os to Windows 11 ,it idisappered.

**Reply by wwwronin** [reply]
I start another discussion you may see later. Multiple imagery layer there is a white aera .

**Reply by gwaldron** [reply]
Rocky doesn't support more than 1 imagery layer yet. It will in the future.
